lib/component-data/mutations.js

```diff
@@ -17,7 +17,14 @@ export default {
     return state;
   },
   [ADD_SCHEMA]: (state, {name, data}) => {
-    _.set(state, `schemas['${name}']`, data);
+    _.set(state, `schemas['${name}']`, {...data, schema_name: name});
```

Comment:
Please mind the spacing
```suggestion
    _.set(state, `schemas['${name}']`, { ...data, schema_name: name });
```

Reply:
👍